setup.py

```diff
@@ -0,0 +1,18 @@
+from skbuild import setup
+
+setup(
+    # NOTE: Could still add stuff like homepage or author mail, but since this isn't used to redistribute, not important
+    name="ezc3d",
+    version="1.4.4", # NOTE: Can this be automatically extracted from CMakeLists.txt without manually regex'ing it out?
```

Comment:
Is this setup file intended to be used before (in replacement) or after the first call ever to CMakeLists? If it is after, you can use configure of CMake to create the setup.py file from a setup.py.in file!
Otherwise, I'll have a look what are the good practice for that case

Reply:
In any case, here is a snippet that retrieve the version from the CMakeLists.txt file assuming the setup.py file is in the same folder. You could replace the 1.4.4 by a function adapted from this?


```python
import re
import os
dir_path = os.path.dirname(os.path.realpath(__file__))

with open(f"{dir_path}/CMakeLists.txt") as file:
    for line in file:
        match = re.search("project\(ezc3d VERSION ([0-9].[0-9].[0-9])\)", line)
        if match is not None:
            version = match[1]
            break
    else:
        raise RuntimeError("Version not found")
print(version)
```

Reply:
The setup file calls CMake, so it's kinda a replacement for CMake.

I added your snippet (Thanks for providing it!)